### PR TITLE
Report errors in beforeAll and afterAll with Jasmine

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

print and report errors in `beforeAll` and `afterAll` hooks.
- Errors are printed when they happen (not in the end of suite)
- `beforeAll` and `afterAll` hook errors in root scope are printed as well

image

image

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works

## Further comments

another approach to deal with #3084 

ref: @martinfrancois  https://github.com/webdriverio/webdriverio/pull/4566

### CODE REVIEW

**Comment by mgrybyk** [comment]
going to add unit tests soon

**Comment by martinfrancois** [comment]
@mgrybyk feel free to copy the unit tests from my branch (from my original PR), the behavior should be the same 😉

**Comment by mgrybyk** [comment]
had to decouple `'should properly set up jasmine'` because it became too huge.

File: tests/jasmine/test.js [hunk 1]
```diff
@@ -1,4 +1,16 @@
+beforeAll(() => {})
+afterAll(() => {
+    throw new Error('afterAll failure in root scope')
```
Comment (christian-bromann) [inline tests/jasmine/test.js]: two questions:

1. shouldn't this fail the smoke tests?
2. if not can we somehow assert the spec report that it outputs proper logs?

Comment (mgrybyk) [reply tests/jasmine/test.js]: 1. beforeAll / afterAll hook failures are not failing the suite
2.1. root scope hooks and tests don't appear in any reporters including spec (for jasmine)
2.2. don't know if we can validate spec reporter result, let me check that

Comment (mgrybyk) [reply tests/jasmine/test.js]: point 2.1 is a defect I think (not related to the PR)

Comment (mgrybyk) [reply tests/jasmine/test.js]: we can validate reporter output with `wdio-smoke-test-reporter`, let me do that

Comment (mgrybyk) [reply tests/jasmine/test.js]: @christian-bromann added smoke test with reporter and raised a defect https://github.com/webdriverio/webdriverio/issues/4582

**Pull request merged**